core/lib/state/src/storage_factory.rs

```diff
@@ -11,19 +11,20 @@ use crate::{
     PostgresStorage, ReadStorage, RocksdbStorage, RocksdbStorageBuilder, StateKeeperColumnFamily,
 };
 
-/// Factory that can produce [`OwnedStorage`] instances on demand.
+/// Factory that can produce storage instances on demand. The storage type is encapsulated as a type param
+/// (mostly for testing purposes); the default is [`OwnedStorage`].
 #[async_trait]
-pub trait ReadStorageFactory: Debug + Send + Sync + 'static {
-    /// Creates an [`OwnedStorage`] entity over either a Postgres connection or RocksDB
-    /// instance. The specific criteria on which one are left up to the implementation.
+pub trait ReadStorageFactory<S = OwnedStorage>: Debug + Send + Sync + 'static {
```

Comment:
I'm not sure whether this should be a type param or an associated type. Approach with the type param allows to define a default (i.e., less code changes), so I've chosen it.

Reply:
Looks fine to me.